foo_spider_monkey_panel/panel/js_panel_window.cpp

```diff
@@ -306,24 +306,46 @@ void js_panel_window::ExecuteEvent_Basic( EventId id )
         {
             break;
         }
+
+        // from ms-docs: The update rectange retrieved by GetUpdateRect
+        // is identical to the one retrieved by the BeginPaint function.
+
+        // BeginPaint automatically validates the update region, so any
+        // call to GetUpdateRect made immediately after the call to BeginPaint
+        // retrieves an empty update region.
+
+        // skip redraw if the update rectangle is empty
+        CRect updateRect;
+        wnd_.GetUpdateRect( &updateRect, FALSE );
```

Comment:
this might bork the repaint, since `isBgRepaintNeeded_` won't be reset to false.

Reply:
hmm, that might be true, however the bg repaint would still be needed as it didn't really happen (?)

so now that i looked at the code / msdn docs for a while, it seems that `isBgRepaintNeeded` is set by the next case item (which is proxying from WM_ERASEBACKGROUND, and from OnSizeUser), with both cases occuring on a resize event indicating that the updateRect shouldn't be empty at all in these cases.
giving some mre tought, setting the flag here to false couldn't hurt (in the case of resizing the panel to 0 width/height)